source/code/plugins/in_dsc_monitor.rb

```diff
@@ -85,6 +86,7 @@ def run_check_status
           @dsc_cache[:status] = dsc_status
           @dsc_cache.commit
         }
+        File.chmod(0644, dsc_cache_file)
```

Comment:
I tested it that way, but the file was not actually created there. I've added it here because the file is guaranteed to exist there, and it is after the YAML Store class does whatever they do to edit that file.